tests/e2e/e2e_test.go

```diff
@@ -202,6 +213,11 @@ func (s *IntegrationTestSuite) TestTWAP() {
 		s.T().Logf("Current epoch number is (%d), waiting to reach (%d)", newEpochNumber, oldEpochNumber+1)
 		return newEpochNumber > oldEpochNumber
 	})
+
+	// We should still be able to get the TWAP after pruning.
+	// It should not change from the post-swap TWAP.
+	twapToNowPostPruning := chainANode.QueryGetArithmeticTwap(poolId, denomOne, denomTwo, time.Now())
+	s.Require().Equal(twapToNowPostPruning, twapToNowPostSwap)
```

Comment:
I'm currently trying to figure out why these would not be equal.

My thinking is that they should still be since they are supposed to utilize the same start record

Reply:
Ah right this should be addressed in https://github.com/osmosis-labs/osmosis/pull/2635

Reply:
Yeah, we have a bug if they're not equal. (Though this line uses time.Now() right above)

Reply:
You are right, `time.Now()` was the problem, thanks. The new check you added exposed the invalid setup. Then, making time be at fixed intervals from queried block times helped fix this.

All tests are passing right now. I'm going to make a 3-asset pool next to try and repro the bug found in the review